tests/data_context/test_data_context.py

```diff
@@ -1908,3 +1909,34 @@ def test_get_validator_with_attach_expectation_suite(
         create_expectation_suite_with_name="A_expectation_suite",
     )
     assert my_validator.expectation_suite_name == "A_expectation_suite"
+
+
+def test_get_batch_multiple_datasources_do_not_scan_all(
+    data_context_with_bad_datasource,
+):
+    """
+    What does this test and why?
```

Comment:
@Shinnnyshinshin So basically, the stale/misconfigured data sources can exist, but they will not harm us; is this the enhancement?